src/PeoplePage.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useState } from 'react';
+import { EnumPersonType } from './types/EnumPersonType';
+import { PersonLink } from './PersonLink';
+import { Person } from './types';
+
+interface PeoplePageProps {
+  usersArr: Person[];
+}
+
+export const PeoplePage:React.FC<PeoplePageProps> = ({ usersArr }) => {
+  const [selectedPerson, setSelectedPerson] = useState<string | null>(null);
+
+  const handlePersonClick = (personName: string | null) => {
+    setSelectedPerson(personName);
+  };
+
+  const isPresentInTable = (name: string | null) => {
+    return usersArr.some((person) => person.name === name);
```

Comment:
The isPresentInTable function is reiterating over the usersArr on every call. This could lead to performance issues when the array is large. Instead, consider using a lookup object to map person names to their existence in the array.

Reply:
@lerastarynets I am not sure I understand what lookup obj is and how I write it and what it improves here